### PR TITLE
feat(http/unstable): `headers` module

### PR DESCRIPTION
**Pull request by halvardssm**
Adds IANA Headers as a follow-up from #4304

### CODE REVIEW

File: http/header.ts [hunk 1]
```diff
@@ -0,0 +1,1685 @@
...
+ * @see {@link https://www.iana.org/assignments/http-fields/http-fields.xhtml#field-names | IANA Hypertext Transfer Protocol (HTTP) Field Name Registry}
+ */
+export const HttpHeader = {
+  ...HttpHeaderPermanent,
+  ...HttpHeaderDeprecated,
+  ...HttpHeaderObsoleted,
+  ...HttpHeaderProvisional,
+};
+
+export type HttpHeader = typeof HttpHeader[keyof typeof HttpHeader];
```
Comment (iuioiua) [inline http/header.ts:1685]:
```suggestion
export type Header = typeof HEADER[keyof typeof HEADER];
```

File: http/header.ts [hunk 2]
```diff
@@ -0,0 +1,1685 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+// This module is generated from {@link https://www.iana.org/assignments/http-fields/http-fields.xhtml#field-names | IANA Hypertext Transfer Protocol (HTTP) Field Name Registry}
+
+/**
+ * HTTP Headers with status permanent
+ */
+export const HttpHeaderPermanent = {
```
Comment (iuioiua) [inline http/header.ts:8]:
```suggestion
export const HEADER = {
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Yoshiya and I would now like to have this addition.

Can you please remove `HttpHeaderDeprecated`, `HttpHeaderObsoleted` and `HttpHeaderProvisional` objects? What's now `HttpHeaderPermanent` seems to be the only object needed for now.

Can you please also add simple/minimal tests to ensure the plumbing works?

**Comment by halvardssm** [comment]
@iuioiua Done! Let me know if there is anything else!

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you for this.

**Pull request merged**